src/telemetry-tracing.ts

```diff
@@ -233,14 +237,27 @@ export class PubsubSpans {
       [SemanticAttributes.MESSAGING_DESTINATION]: topicName,
       [SemanticAttributes.MESSAGING_DESTINATION_KIND]: 'topic',
       [SemanticAttributes.MESSAGING_PROTOCOL]: 'pubsub',
-      [SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES]:
-        message.data?.length,
-      'messaging.pubsub.ordering_key': message.orderingKey,
     } as SpanAttributes;
 
+    if (message) {
+      if (message.data?.length) {
+        spanAttributes[
+          SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES
+        ] = message.data?.length;
+      }
+      if (message.orderingKey) {
+        spanAttributes['messaging.gcp.pubsub.ordering_key'] =
```

Comment:
```suggestion
        spanAttributes['messaging.gcp_pubsub.message.ordering_key'] =
```

[messaging.gcp_pubsub.message.ordering_key](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/messaging.md)